bin/preprocess.ml

```diff
@@ -0,0 +1,85 @@
+open Stdune
+open Import
+
+let doc = "Print the preprocessed output"
+
+let man =
+  [ `S "DESCRIPTION"
+  ; `P
+      {|$(b,dune preprocess FILE) build the given FILE and print the preprocessed output|}
+  ; `Blocks Common.help_secs
+  ]
+
+let info = Term.info "preprocess" ~doc ~man
+
+let change_ext f path =
+  let base, ext = Path.of_string path |> Path.split_extension in
+  let suffix = f ext in
+  Path.extend_basename base ~suffix |> Path.to_string
+
+let pp_with_ocamlc sctx project pp_ml =
+  let open Dune_engine in
+  let ocamlc = (Super_context.context sctx).ocamlc in
+  let env = Super_context.context_env sctx in
+  let argv = ["-stop-after"; "parsing"; "-dsource"; pp_ml; "-dump-into-file"] in
+  let open Fiber.O in
+  let* _ = Process.run ~env Process.Strict ocamlc argv in
```

Comment:
We always annotate the types of values we ignore.

Reply:
I think I can just do `let* () = ...` since the return type is `unit`.